packages/react-priority-overflow/src/useOverflowItem.ts

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import { useOverflowContext } from './overflowContext';
+
+/**
+ * Registers an overflow item
+ * @param id - unique identifier for the item used by the overflow manager
+ * @param priority - higher priority means the item overflows later
+ * @param groupId - assigns the item to a group, group visibility can be watched
+ * @returns ref to assign to an intrinsic HTML element
+ */
+export function useOverflowItem<TElement extends HTMLElement>(
+  id: string | number,
+  priority?: number,
+  groupId?: string | number,
+) {
+  const ref = React.useRef<TElement>(null);
+  const registerItem = useOverflowContext(v => v.registerItem);
+
+  useIsomorphicLayoutEffect(() => {
+    let deregisterItem: () => void = () => null;
+    if (ref.current) {
+      deregisterItem = registerItem({
+        element: ref.current,
+        id: id + '',
+        priority: priority ?? 0,
+        groupId: groupId + '',
```

Comment:
Casting to string everywhere is annoying 🙃 Did you consider to allow only strings in API?